src/ServicePulse.Host/vue/src/components/DataView.vue

```diff
@@ -0,0 +1,173 @@
+<script setup>
+import { computed, onMounted, onUnmounted, ref } from "vue";
+import { useFetchFromServiceControl } from "../composables/serviceServiceControlUrls";
+
+const props = defineProps({
+  apiUrl: String,
+  itemsPerPageOptions: {
+    type: Array,
+    default: () => [20, 35, 50, 75],
+  },
+  itemsPerPage: {
+    type: Number,
+    default: 50,
+  },
+  autoRefresh: {
+    type: Number,
+    default: 0,
+  },
+  showPagination: {
+    type: Boolean,
+    default: true,
+  },
+  showItemsPerPage: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const refreshTimer = ref();
+const items = ref([]);
+const pageNumber = ref(1);
+const itemsPerPage = ref(props.itemsPerPage);
+const totalCount = ref(0);
+const showPagination = computed(() => {
+  return props.showPagination && numberOfPages.value > 1;
+});
+
+const numberOfPages = computed(() => {
+  return Math.ceil(totalCount.value / itemsPerPage.value);
+});
+
+const showItemsPerPage = computed(() => {
+  return props.showItemsPerPage;
+});
+
+function changeItemsPerPage(value) {
+  itemsPerPage.value = value;
+  loadData();
+}
+
+function calculatePageNumbers() {
+  if (numberOfPages.value <= 10) {
+    return Array.from(Array(numberOfPages.value), (_, index) => index + 1);
```

Comment:
i can see what this is doing, but it's confusing and needs a comment. Also, supposedly faster to have Array(numberOfPages.value).fill(0).map((_, i) => i + 1)

Reply:
This was completely re-written